lambda/src/main/scala/pricemigrationengine/handlers/EstimationHandler.scala

```diff
@@ -32,6 +32,36 @@ object EstimationHandler extends CohortHandler {
         .tapError(e => Logging.error(e.toString))
     } yield HandlerOutput(isComplete = count < batchSize)
 
+  def monitorDoNotProcessUntilItem(today: LocalDate, item: CohortItem): ZIO[Logging with CohortTable, Failure, Unit] = {
+    for {
+      _ <-
+        if (CohortItem.isProcessable(item, today)) {
+          CohortTable
+            .update(
+              CohortItem(
+                subscriptionName = item.subscriptionName,
+                processingStage = ReadyForEstimation
+              )
+            )
+        } else { ZIO.succeed(()) }
+    } yield ()
+  }
+
+  def monitorDoNotProcessUntils(today: LocalDate): ZIO[Logging with CohortTable, Failure, Unit] = {
+    // This function migrates items in DoNotProcessUntil state, which have passed their
+    // expiration time, to ReadyForEstimation
+    for {
+      _ <- CohortTable
+        .fetch(DoNotProcessUntil, None)
```

Comment:
is there a date filter for this fetch call?

Reply:
@tomrf1 

Good question. Answer is No. 

That function is defined here: https://github.com/guardian/price-migration-engine/blob/db76ee8627aa1b7732f3479c92c7a3f9792ddff8/lambda/src/main/scala/pricemigrationengine/services/CohortTableLive.scala#L127 and as you can see it either (if the second argument is `None`) retrieves all the items of a given processing stage or (if the second argument is defined) all items of a given processing stage whose `startDate` is before a given date. 

Most times it's used, for instance here https://github.com/guardian/price-migration-engine/blob/db76ee8627aa1b7732f3479c92c7a3f9792ddff8/lambda/src/main/scala/pricemigrationengine/handlers/AmendmentHandler.scala#L22 , we do not specify a date because we want all the items. 

In this case I want all the items in `DoNotProcessUntil`, and this function https://github.com/guardian/price-migration-engine/pull/1056/files#diff-23bf94a527ab471ddb3378219c90b6bc237922cbc47e53ea002e247b611a5e6eR35 will decide what to do with individual items.